amazon-kinesis-client/src/main/java/software/amazon/kinesis/coordinator/Scheduler.java

```diff
@@ -193,12 +193,14 @@ public void run() {
             return;
         }
 
+        Exception exceptionToThrow = null;
```

Comment:
This doesn't feel like the right place for this.  Should we instead put onInitializationException at the point that the exception was thrown?
This just reports the last exception, which means a cascaded failure may be misreported.